src/shogun/mathematics/linalg/LinalgBackendViennacl.h

```diff
@@ -115,6 +186,27 @@ class LinalgBackendViennaCL : public LinalgBackendGPUBase
 		return viennacl::linalg::inner_prod(a_gpu->data(a.size()), b_gpu->data(b.size()));
 	}
 
+	/** Return the largest element in the vector with ViennaCL library */
+	template <typename T>
+	T max_impl(const SGVector<T>& vec) const
+	{
+		viennacl::ocl::kernel& kernel = generate_kernel<T>();
```

Comment:
@lambday so this kernel is compiled everytime we call this operation?
Is that good?